MIGRATION.md

```diff
@@ -0,0 +1,264 @@
+</aside>
+
+## 1. 60% size reduction
+
+Through simplifying 3rd party dependencies, xrpl.js is now a much ligher-weight library.
+
+One major source of the project’s large bundle size was polyfills, code that replicated Node specific features in the browser. Instead, we shifted to using 3rd party packages which worked in both Node and the browser by default.
+
+A simple fix was removing `lodash` by using es6 array methods and porting over simple helper utilities.
+
+The last main source of the reduction was simplifying the 5 large number libraries down to one.
+
+Before, we had `decimal.js`, `bit-integer`, `bitnumber.js`, and two versions of `bn.js` due to `elliptic`'s transitive dependency tree.
+
+After, we used `@noble` to replace `elliptic` to have just one version of `bn.js`, and normalized our library to just use `bignumber.js` across the board.
+
+## 2. No more polyfills required (simplified install)
+
+Polyfills made it hard to setup xrpl.js in the browser as they required custom bundler configs. By using dependencies and browser-native features, xrpl.js can now work just by installing from `npm` in most cases.
+
+For the cryptography libraries, we switched from using `elliptic`, `create-hash`, and other crypto polyfills to using the `@noble` suite of packages. For situations where node-specific crypto functions performed better, we created `@xrplf/isomorphic` to dynamically choose which implementation to use depending on the runtime environment.
+
+<aside>
+💡 The `@noble` suite of packages includes `@noble/hashes`, `@noble/curves`, `@scure/bip32`, `@scure/bip39`, and `@scure/base`
+
+</aside>
+
+We eliminated the polyfills for `http`, `https`, and `url` by using the native `fetch` in the browser.
+
+The easiest to replace were `assert` (which was replaced by simple conditions & exceptions) and `utils` (which used `JSON.stringify` instead of `inspect`).
+
+Lastly, the `buffer` polyfill turned out to be the trickiest to remove, resulting in the largest number of breaking changes. Since the `Buffer` object is not native to the browser all apis were migrated to the superclass of `Buffer` → `Uint8Arrays`. For a detailed write up of why we and many libraries are choosing to make this transition, check out this [blog post](https://sindresorhus.com/blog/goodbye-nodejs-buffer) by Sindre Sorhus.
+
+## 3. Increased Reliability
+
+With xrpl.js 3.0, we improved our test coverage in the browser. Specifically, we added browser unit testing to all packages in the monorepo other than the `xrpl` package. Note that the `xrpl` package has browser coverage through our integration tests.
+
+We achieved this by adding a `karma` config to every project that uses webpack to bundle each lib, then run the tests in Chrome. Hopefully soon we’ll also be able to add support for the `xrpl` package to run it’s unit tests in Chrome as part of our CI process.
```

Comment:
In Xrpl.js 3.0, we enhanced our testing capabilities for browser compatibility. This improvement involved the introduction of browser unit testing for all packages within the monorepo, except for the xrpl package, which already had browser coverage through integration tests.
```suggestion
With xrpl.js 3.0, we improved our test coverage in the browser. Specifically, we added browser unit testing to all packages in the monorepo other than the `xrpl` package. Note that the `xrpl` package has browser coverage through our integration tests.

We achieved this by adding a `karma` config to every project that uses webpack to bundle each lib, then run the tests in Chrome. Hopefully soon we’ll also be able to add support for the `xrpl` package to run it’s unit tests in Chrome as part of our CI process.
```

To implement this enhancement, we included a karma configuration in every project utilizing webpack to bundle each library, allowing us to execute tests in the Chrome browser. We are actively working towards extending this support to include running unit tests for the xrpl package in Chrome as an integral part of our continuous integration (CI) process in the near future.